satpy/readers/iasi_ng_l2_nc.py

```diff
@@ -0,0 +1,418 @@
+
+    @property
+    def end_time(self):
+        """Get the end time."""
+        return self.filename_info["sensing_end_time"]
+
+    @property
+    def sensor_names(self):
+        """List of sensors represented in this file."""
+        return self.sensors
+
+    # Note: patching the collect_groups_info method below to
+    # also collect dimensions in sub groups.
+    def _collect_groups_info(self, base_name, obj):
+        for group_name, group_obj in obj.groups.items():
+            full_group_name = base_name + group_name
+            self.file_content[full_group_name] = group_obj
+            self._collect_attrs(full_group_name, group_obj)
+            self.collect_metadata(full_group_name, group_obj)
+            self.collect_dimensions(full_group_name, group_obj)
+
+    def available_datasets(self, configured_datasets=None):
+        """Determine automatically the datasets provided by this file.
+
+        Uses a per product type dataset registration mechanism.
+        """
+        # pass along existing datasets
+        for is_avail, ds_info in configured_datasets or []:
+            yield is_avail, ds_info
+
+        for _, ds_info in self.dataset_infos.items():
+            yield True, ds_info
+
+    def register_dataset(self, ds_name, desc):
+        """Register a simple dataset given its name and a desc dict."""
+        if ds_name in self.dataset_infos:
+            raise KeyError(f"Dataset for {ds_name} already registered.")
+
+        ds_infos = {
+            "name": ds_name,
+            "sensor": "iasi_ng",
+            "file_type": self.filetype_info["file_type"],
+        }
+
+        ds_infos.update(desc)
+
+        self.dataset_infos[ds_name] = ds_infos
+
+    def process_dimension(self, key, value):
+        """Process a dimension entry from the file_content."""
+        dim_name = key.split("/")[-1]
+
+        if dim_name in self.dimensions_desc and self.dimensions_desc[dim_name] != value:
+            # This might happen if we have the same dim name from different groups:
+            raise KeyError(f"Detected duplicated dim name: {dim_name}")
+
+        self.dimensions_desc[dim_name] = value
+
+    def process_attribute(self, key, value):
+        """Process a attribute entry from the file_content."""
+        var_path, aname = key.split("/attr/")
+
+        if var_path not in self.variable_desc:
+            # maybe this variable is ignored, or this is a group attr.
+            return
+
+        self.variable_desc[var_path]["attribs"][aname] = value
+
+    def process_variable(self, key):
+        """Process a variable entry from the file_content."""
+        shape = self.file_content[f"{key}/shape"]
+
+        if np.prod(shape) <= 1:
+            # Ignoring scalar variable.
+            return
+
+        # Check if this variable should be ignored:
+        if any(p.search(key) is not None for p in self.ignored_patterns):
+            # Ignoring variable on user request.
```

Comment:
if you put this condition in its own function that is called eg `variable_should_be_ignored`, then you don’t need comments.